openstf/model/fallback.py

```diff
@@ -0,0 +1,68 @@
+# SPDX-FileCopyrightText: 2017-2021 Alliander N.V. <korte.termijn.prognoses@alliander.com> # noqa E501>
+#
+# SPDX-License-Identifier: MPL-2.0
+from pathlib import Path
+from datetime import datetime, timedelta, timezone
+
+import pandas as pd
+import structlog
+
+
+def generate_fallback(
+    forecast_input: pd.DataFrame,
+    load: pd.DataFrame,
+    fallback_strategy: str = "extreme_day",
+) -> pd.DataFrame:
+    """Make a fall back forecast,
+    Set the value of the forecast 'quality' column to 'substituted'
+
+    Currently only fallback_strategy=extreme day is implemented which return historic profile of most extreme day.
+
+    Args:
+        forecast_input (pandas.DataFrame): dataframe desired for the forecast
+        load (pandas.DataFrame): index=datetime, columns=['load']
+        fallback_strategy (str): strategy to determine fallback. options:
+            - extreme_day: use daily profile of most extreme day
+    Returns:
+        pandas.DataFrame: Fallback forecast DataFrame with columns:
+            'forecast', 'quality'
+    """
+    # Check if load is completely empty
+    if len(load.dropna()) == 0:
+        raise ValueError("No historic load data available")
+
+    if fallback_strategy != "extreme_day":
+        raise NotImplementedError(
+            f'fallback_strategy should be "extreme_day", received:{fallback_strategy}'
+        )
+
+    # Find most extreme historic day (do not count today as it is incomplete)
+    day_with_highest_load_date = (
+        load[load.index.tz_localize(None).date != datetime.utcnow().date()]
+        .idxmax()
+        .load.date()
+    )
+    # generate datetime range of the day with the highest load
+    from_datetime = pd.Timestamp(day_with_highest_load_date, tz=load.index.tz)
+    till_datetime = from_datetime + pd.Timedelta("1 days")
+
+    # slice load dataframe, all rows for the day with the highest load
+    highest_daily_loadprofile = load.loc[
+        (load.index >= from_datetime) & (load.index < till_datetime)
+    ]
+
+    highest_daily_loadprofile.loc[:, "time"] = highest_daily_loadprofile.index.time
+
+    forecast = pd.DataFrame(index=forecast_input)
+    forecast["time"] = forecast.index.time
+    forecast = (
```

Comment:
I could not really tell what exactly happens here, but that's probably my lack of pandas knowledge. If its easier to make it more clear (maybe just by a comment) that could improve readability perhaps.

Reply:
I've added some comments in the function to make it (hopefully) a little more clear.